legate/driver/command.py

```diff
@@ -0,0 +1,363 @@
+
+
+def cmd_kthreads(
+    config: Config, system: System, launcher: Launcher
+) -> CommandPart:
+    freeze_on_error = config.debugging.freeze_on_error
+    gdb = config.debugging.gdb
+    cuda_gdb = config.debugging.cuda_gdb
+
+    if freeze_on_error or gdb or cuda_gdb:
+        # Running with userspace threads would not allow us to inspect the
+        # stacktraces of suspended threads.
+        return ("-ll:force_kthreads",)
+
+    return ()
+
+
+def cmd_cpus(
+    config: Config, system: System, launcher: Launcher
+) -> CommandPart:
+    cpus = config.core.cpus
+
+    return () if cpus == 1 else ("-ll:cpu", str(cpus))
+
+
+def cmd_gpus(
+    config: Config, system: System, launcher: Launcher
+) -> CommandPart:
+    gpus = config.core.gpus
+
+    # Make sure that we skip busy GPUs
+    return () if gpus == 0 else ("-ll:gpu", str(gpus), "-cuda:skipbusy")
+
+
+def cmd_openmp(
+    config: Config, system: System, launcher: Launcher
+) -> CommandPart:
+    openmp = config.core.openmp
+    ompthreads = config.core.ompthreads
+    numamem = config.memory.numamem
+
+    if openmp == 0:
+        return ()
+
+    if ompthreads == 0:
+        print(
+            warn(
+                f"Legate is ignoring request for {openmp} "
+                "OpenMP processors with 0 threads"
+            )
+        )
+        return ()
+
+    return (
+        "-ll:ocpu",
+        str(openmp),
+        "-ll:othr",
+        str(ompthreads),
+        # If we have support for numa memories then add the extra flag
+        "-ll:onuma",
+        f"{int(numamem > 0)}",
+    )
+
+
+def cmd_utility(
+    config: Config, system: System, launcher: Launcher
+) -> CommandPart:
+    utility = config.core.utility
+    ranks = config.multi_node.ranks
+
+    if utility == 1:
+        return ()
+
+    opts: CommandPart = ("-ll:util", str(utility))
+
+    # If we are running multi-rank then make the number of active
+    # message handler threads equal to our number of utility
+    # processors in order to prevent head-of-line blocking
+    if ranks > 1:
+        opts += ("-ll:bgwork", str(utility))
+
+    return opts
+
+
+def cmd_mem(config: Config, system: System, launcher: Launcher) -> CommandPart:
+    # Always specify the csize
+    return ("-ll:csize", str(config.memory.sysmem))
+
+
+def cmd_numamem(
+    config: Config, system: System, launcher: Launcher
+) -> CommandPart:
+    numamem = config.memory.numamem
+    return () if numamem == 0 else ("-ll:nsize", str(numamem))
+
+
+def cmd_fbmem(
+    config: Config, system: System, launcher: Launcher
+) -> CommandPart:
+    if config.core.gpus == 0:
+        return ()
+
+    fbmem, zcmem = config.memory.fbmem, config.memory.zcmem
+    return ("-ll:fsize", str(fbmem), "-ll:zsize", str(zcmem))
+
+
+def cmd_regmem(
+    config: Config, system: System, launcher: Launcher
+) -> CommandPart:
+    regmem = config.memory.regmem
+    return () if regmem == 0 else ("-ll:rsize", str(regmem))
+
+
+def cmd_log_levels(
```

Comment:
I think the user-selected --logging levels should be added here as well.

Reply:
yikes I think I lost this entire block

https://github.com/nv-legate/legate.core/blob/d299b86296e143c456cb96cca1d8b60d14a872aa/legate/__main__.py#L668-L671

Reply:
@manopapad https://github.com/nv-legate/legate.core/pull/371/commits/6ff9221fca7981ebef25e32144e65e47fcf8dc0e 

let me know if that is enough testing, currently it tests each input to the total `--level` value individually, and then one combined test at the end.

Reply:
Looks good